src/RF24LogAbstracts/BaseHandler.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file RF24LogAbstracts/BaseHandler.h
```

Comment:
This file contains now too much. It should contain only definition of the class that is used in ```RF24Logging``` as a log handler.

I suggest the following:
* rename the file from ```RF24LogAbstracts/BaseHandler.h``` into ```RF24LogHandler.h``` 
* also the file should be moved one directory back - directly under ```src``` so it lays down together with other core files of the logging library: RF24Loglevel.h, RF24Logging.h and RF24Logging.cpp
* the class inside should be renamed from ```RF24LogBaseHandler``` to a simple ```RF24LogHandler```. From the unified logging library this class should act like a pure virtual interface. The implementation should be done in classes that implements this interface (extends ```RF24LogHandler```).

Reply:
RF24AbstractHandler inherits from RF24Handler? It was too vague. In the docs I describe this file as the base mechanism for forwarding data to output streams. I can move the file back to src dir and rename it to RF24LogBaseHandler or RF24LogMainHandler (as long as the file/class name transparently differentiate from RF24LogAbstractHandler).

Reply:
Hm, yes, RF24AbstractHandler inherits from RF24Handler (or RF24LogHandler). I see RF24LogHandler class as a pure abstraction for handling log messages. This class declares a few minimal methods which must be implemented correctly in a derived class. 
Now I can see that the **core** of the unified library can have four files:
* RF24Loglevel.h with log levels definitions
* RF24Logging.h with declarations of what is the unified log library API
* RF24Logging.cpp with basic implementation of redirecting messages to RF24LogHandler
* RF24LogHandler.h with a minimal declaration of how log handler should look like

In other directories - **RF24Loggers**, **RF24LogAbstracts** and **handler_ext** - we offer same default and basic log handlers implementations, so the end user doesn't need to reinvent them from scratch. This is very nice.
Of course it should be possible for the developer to have his totally own implementation of **RF24LogHandler** (in other words for some reason he has no interest in RF24Log default implementations). That's why I push to rename the class **RF24LogBaseHandler** into **RF24LogHandler** (the word **Base** should be thrown away as it is not needed and redundant here) and move it to the **src/RF24LogBaseHandler.h** header file. In this way he doesn't need to include something from the **RF24LogAbstracts** directory.

Reply:
> rename the class RF24LogBaseHandler into RF24LogHandler (the word Base should be thrown away as it is not needed and redundant here) and move it to the src/RF24LogBaseHandler.h header file.

I think you raised the question earlier about having a filename different from the class name within it; the only reason this is concerning to me now is because this file is going to be on the root level of the src dir. The word "Base" is only redundant if you know how the library is structured. Trust me, at first glance, it's confusing to see an abstract class inherit from a class of a less descriptive name. Would you be ok with "RF24LogVirtualHandler"?

Reply:
Or why not RF24LogSupervisor because it is the gate that all log massages have to get through?

Reply:
Thanks for your suggestions but among all of them - **RF24LogBaseHandler** sounds the best :) 

>Trust me, at first glance, it's confusing to see an abstract class inherit from a class of a less descriptive name

Actually this is not confusing me. Less descriptive name means that we deal with a higher level of abstraction. I would like to have this class to be called **RF24LogHandler** because it will be defined on the highest possible level of abstraction.
From the core piece of the library the name **RF24Handler** is the top best. It is just a handler definition. No matter what kind of handler: base handler or abstract handler or stream handler,.. it's just a handler which doesn't have any implementation, it tells what methods are required to handle log messages; it is not important here how those methods will be implemented.

What we have now:
```class RF24LogAbstractHandler : public RF24LogBaseHandler```
or what I would like to have:
```class RF24LogAbstractHandler : public RF24LogHandler```
means to me that ```RF24LogAbstractHandler``` is some kind of "first" real implementation of ```RF24LogHandler```. It probably implements some virtual methods from ```RF24LogHandler``` but also can introduce new methods to avoid code duplication for other class that will inherit from it, etc.

Reply:
>Less descriptive name means that we deal with a higher level of abstraction

Ohhhh. Setting aside what I already know about the lib structure, this is not that obvious to me. In terms of programming, I don't think that this is universally true (@avamander can you verify this is a common notion?). Speaking as an English tutor, its not good to assume what your readers know or understand.

>because it will be defined on the highest possible level of abstraction

This is why I had the notion to put the file into the RF24LogAbstracts folder.

Reply:
>Less descriptive name means that we deal with a higher level of abstraction

I just want to add that this is my feeling and understanding of naming. Of course my suggestions of **RF24Handler** is also a kind of abstraction so it could be placed under **RF24LogAbstracts** folder but a better place is to put it directly under **src** directory as it will act then on the top level of handlers abstraction (other words it is an common **handler** interface definition). I could even then remove **RF24LogAbstracts**, **RF24Loggers** and **handler_ext** directories and I will be still able to use the logging library - but then I need to implement my own **RF24Handler** (totally from scratch).

Reply:
I could go either way on the location of the file, but renaming it to what it was... Sorry, I just can't get on-board with that.

I'm also getting linker errors on windows since moving the file back to src folder (this didn't happen when it lived in the abstracts folder), but I'm still trying to figure out the real error (linker error outputs really just seem to point to a place where I should start looking for errors 😞 ).

Reply:
Ok, let it be, however I still think that the class and file should be renamed into ```RF24LogHandler```. Let's see how it behaves now.